### PR TITLE
Feature fix push crash without phone permission

### LINKED ISSUE #1777: Permissions confirmation deadline
If you don't accept the permissions before the first pull the app crashes

### PR DESCRIPTION
**Pull request by idelcano**
Closes #1777  and #1772

### CODE REVIEW

File: app/src/ereferrals/java/org/eyeseetea/malariacare/services/strategies/PushServiceStrategy.java [hunk 1]
```diff
@@ -57,6 +60,13 @@ public PushServiceStrategy(PushService pushService) {
     @Override
     public void push() {
 
+        if (ActivityCompat.checkSelfPermission(PreferencesState.getInstance().getContext(), Manifest.permission.READ_SMS)
```
Comment (xurxodev) [inline app/src/ereferrals/java/org/eyeseetea/malariacare/services/strategies/PushServiceStrategy.java]: Can we encapsulate this logic in Permissions class?

File: app/src/main/java/org/eyeseetea/malariacare/BaseActivity.java [hunk 2]
```diff
@@ -114,13 +114,16 @@ protected void onCreate(Bundle savedInstanceState) {
...
         if (Permissions.processAnswer(requestCode, permissions, grantResults)) {
-            EyeSeeTeaApplication.permissions.requestNextPermission();
+            EyeSeeTeaApplication.permissions.requestNextPermission(this);
             if (EyeSeeTeaApplication.permissions.areAllPermissionsGranted()) {
                 PhoneMetaData phoneMetaData = getPhoneMetadata();
                 Session.setPhoneMetaData(phoneMetaData);
             }
         } else {
-            onDestroy();
+            Intent homeIntent = new Intent(Intent.ACTION_MAIN);
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/BaseActivity.java:123]: I think that this code provoke a activity refresh first time when you navigate to dashboard from login

**Review by xurxodev — CHANGES_REQUESTED** [review]
@idelcano I have requested you some changes

File: app/src/main/java/org/eyeseetea/malariacare/BaseActivity.java (hunk 2 above)
Comment (idelcano) [reply app/src/main/java/org/eyeseetea/malariacare/BaseActivity.java:123]: i tested it and this code is only executed when the user don't accept the permissions

**Review by xurxodev — APPROVED** [review]
thanks @idelcano

**Pull request merged**